systemvm/debian/opt/cloud/bin/cs/CsAddress.py

```diff
@@ -693,11 +693,8 @@ def post_config_change(self, method):
                 vpccidr = cmdline.get_vpccidr()
                 self.fw.append(
                     ["filter", 3, "-A FORWARD -s %s ! -d %s -j ACCEPT" % (vpccidr, vpccidr)])
-                self.fw.append(
-                    ["nat", "", "-A POSTROUTING -j SNAT -o %s --to-source %s" % (self.dev, self.address['public_ip'])])
-            elif cmdline.get_source_nat_ip() and not self.is_private_gateway():
-                self.fw.append(
-                    ["nat", "", "-A POSTROUTING -j SNAT -o %s --to-source %s" % (self.dev, cmdline.get_source_nat_ip())])
+            self.fw.append(
```

Comment:
if there are multiple public ips (in multiple ranges), will there  be same amount of rules ?

Reply:
I am not sure I understand the question. I checked this in a lab env and the resulting nat table was exactly as described in the issue, with only the last line being different. Ar you considdering another configuration here @weizhouapache ?

Reply:
@DaanHoogland 
private gateway is considered as a public ip as well. 

I had another look at the code, we should 
- keep the code around line 557.
- in line 696, add a check like below

```suggestion
            if self.address["source_nat"] or not self.is_private_gateway():
              self.fw.append(
                  ["nat", "", "-A POSTROUTING -j SNAT -o %s --to-source %s" % (self.dev, self.address['public_ip'])])
```

Reply:
for each public ip (and private gateway), there will be a rule below, right ?
```
-A POSTROUTING -j SNAT -o ethX --to-source xx.yy.zz.xx
```

Reply:
@DaanHoogland 
to be clear, we need a rule for each public NIC, for example

```
-A POSTROUTING -j SNAT -o eth1 --to-source <source nat IP>    # this is for source nat NIC
-A POSTROUTING -j SNAT -o eth5 --to-source <first public IP on eth5>    # this is for additional public NIC
```

If I understand correctly, for the current changes , the rules are for example,
```
-A POSTROUTING -j SNAT -o eth1 --to-source <source nat IP>    # this is for source nat NIC
-A POSTROUTING -j SNAT -o eth1 --to-source <second IP on source nat NIC>    # this is for source nat NIC
-A POSTROUTING -j SNAT -o eth1 --to-source <third IP on source nat NIC>    # this is for source nat NIC

-A POSTROUTING -j SNAT -o eth5 --to-source <first public IP on eth5>    # this is for additional public NIC
-A POSTROUTING -j SNAT -o eth5 --to-source <second public IP on eth5>    # this is for additional public NIC
-A POSTROUTING -j SNAT -o eth5 --to-source <third public IP on eth5>    # this is for additional public NIC
```

Reply:
I'll verify that. Do you happen to know what condition to test for? I don't think the `self.address` object contains information on whether it is the first IP, does it?

Reply:
I think the original issue does not exist in our lab (I can verify with infra). 

we can only verify the iptables rules in the VR
- create 2 public ip ranges with different vlan
- acquire 3 public ips on each public ip and use them for static/pf/lb

Reply:
> I don't think the self.address object contains information on whether it is the first IP, does it?

it does, but I am not sure if it is 100% correct.

![image](https://github.com/user-attachments/assets/369ff4db-9d61-4a87-a166-2a98ac482e4e)

Reply:
ok, I'll give it a try